shell/platform/android/io/flutter/plugin/platform/SingleViewPresentation.java

```diff
@@ -0,0 +1,482 @@
+    rootView.addView(container);
+    rootView.addView(state.fakeWindowViewGroup);
+
+    embeddedView.setOnFocusChangeListener(focusChangeListener);
+    rootView.setFocusableInTouchMode(true);
+    if (startFocused) {
+      embeddedView.requestFocus();
+    } else {
+      rootView.requestFocus();
+    }
+    setContentView(rootView);
+  }
+
+  public PresentationState detachState() {
+    container.removeAllViews();
+    rootView.removeAllViews();
+    return state;
+  }
+
+  public PlatformView getView() {
+    if (state.platformView == null) return null;
+    return state.platformView;
+  }
+
+  /*
+   * A view group that implements the same layout protocol that exist between the WindowManager and its direct
+   * children.
+   *
+   * Currently only a subset of the protocol is supported (gravity, x, and y).
+   */
+  static class FakeWindowViewGroup extends ViewGroup {
+    // Used in onLayout to keep the bounds of the current view.
+    // We keep it as a member to avoid object allocations during onLayout which are discouraged.
+    private final Rect viewBounds;
+
+    // Used in onLayout to keep the bounds of the child views.
+    // We keep it as a member to avoid object allocations during onLayout which are discouraged.
+    private final Rect childRect;
+
+    public FakeWindowViewGroup(Context context) {
+      super(context);
+      viewBounds = new Rect();
+      childRect = new Rect();
+    }
+
+    @Override
+    protected void onLayout(boolean changed, int l, int t, int r, int b) {
+      for (int i = 0; i < getChildCount(); i++) {
+        View child = getChildAt(i);
+        WindowManager.LayoutParams params = (WindowManager.LayoutParams) child.getLayoutParams();
+        viewBounds.set(l, t, r, b);
+        Gravity.apply(
+            params.gravity,
+            child.getMeasuredWidth(),
+            child.getMeasuredHeight(),
+            viewBounds,
+            params.x,
+            params.y,
+            childRect);
+        child.layout(childRect.left, childRect.top, childRect.right, childRect.bottom);
+      }
+    }
+
+    @Override
+    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
+      for (int i = 0; i < getChildCount(); i++) {
+        View child = getChildAt(i);
+        child.measure(atMost(widthMeasureSpec), atMost(heightMeasureSpec));
+      }
+      super.onMeasure(widthMeasureSpec, heightMeasureSpec);
+    }
+
+    private static int atMost(int measureSpec) {
+      return MeasureSpec.makeMeasureSpec(MeasureSpec.getSize(measureSpec), MeasureSpec.AT_MOST);
+    }
+  }
+
+  /** Answers calls for {@link InputMethodManager} with an instance cached at creation time. */
+  // TODO(mklim): This caches the IMM at construction time and won't pick up any changes. In rare
+  // cases where the FlutterView changes windows this will return an outdated instance. This
+  // should be fixed to instead defer returning the IMM to something that know's FlutterView's
+  // true Context.
```

Comment:
This should have a bug filed and linked.

Reply:
could we do this in a separate PR?

Reply:
https://github.com/flutter/flutter/issues/106443